StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1506CodeFixProvider.cs

```diff
@@ -52,32 +52,50 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             var triviaList = token.LeadingTrivia;
 
             var index = triviaList.IndexOf(SyntaxKind.SingleLineDocumentationCommentTrivia);
-            for (; index < triviaList.Count - 1; index++)
+
+            int currentLineStart = index + 1;
+            bool onBlankLine = true;
+            for (int currentIndex = currentLineStart; currentIndex < triviaList.Count; currentIndex++)
             {
-                if (triviaList[index].IsKind(SyntaxKind.SingleLineCommentTrivia)
-                    || triviaList[index].IsKind(SyntaxKind.MultiLineCommentTrivia))
+                switch (triviaList[currentIndex].Kind())
                 {
-                    break;
-                }
-            }
+                case SyntaxKind.EndOfLineTrivia:
+                    if (onBlankLine)
+                    {
+                        for (int i = 0; i < currentIndex - currentLineStart + 1; i++)
+                        {
+                            triviaList = triviaList.RemoveAt(currentLineStart);
+                            currentIndex = currentLineStart - 1;
+                        }
```

Comment:
The assignment to `currentIndex` should have appeared outside the loop. I replaced the loop with a call to a new extension method `RemoveRange` to help clarify the code.